resources/test-invoker.mjs

```diff
@@ -48,8 +67,51 @@ export class RAFTestInvoker extends TestInvoker {
     }
 }
 
+class AsyncRAFTestInvoker extends BaseRAFTestInvoker {
+    _scheduleCallbacks(resolve) {
+        requestAnimationFrame(async () => {
+            await this._syncCallback();
+
+            let gotTimer = false;
+            let gotMessage = false;
+
+            const tryTriggerAsyncCallback = () => {
+                if (!gotTimer || !gotMessage)
+                    return;
+
+                this._asyncCallback();
+                setTimeout(async () => {
+                    await this._reportCallback();
+                    resolve();
+                }, 0);
+            };
+
+            setTimeout(() => {
+                gotTimer = true;
+                tryTriggerAsyncCallback();
+            });
+
+            const mc = new MessageChannel();
+            mc.port1.onmessage = () => {
+                mc.port1.close();
+                mc.port2.close();
+
+                gotMessage = true;
+                tryTriggerAsyncCallback();
+            };
+            mc.port2.postMessage("speedometer");
+        });
+    }
+}
+
 export const TEST_INVOKER_LOOKUP = {
     __proto__: null,
     timer: TimerTestInvoker,
     raf: RAFTestInvoker,
 };
+
+export const ASYNC_TEST_INVOKER_LOOKUP = {
+    __proto__: null,
+    timer: AsyncTimerTestInvoker,
```

Comment:
If we don't want an AsyncTimerTestInvoker, I could simply assign the same AsyncRAFTestInvoker..